### DISCUSSION TITLE
Spacingd cause kernel crashing

### DISCUSSION THREAD

**Discussion by emi-dm** [Project-MONAI/MONAI#7473] [General]
Certainly! Here's a draft for your GitHub issue:

---

**Issue Title**: Bug in Monai Spacingd Transformation

**Description**:
I encountered an issue with the Monai library related to the `Spacingd` transformation. However, I'm unsure if it's due to an error on my part. I've also provided a notebook to reproduce the issue.

**Steps to Reproduce**:
You can reproduce it in the notebook: [https://colab.research.google.com/drive/1ahj3IQqetxRxeVQm0tNpdqXJOq9nTxRT?usp=sharing](https://colab.research.google.com/drive/1ahj3IQqetxRxeVQm0tNpdqXJOq9nTxRT?usp=sharing)

**Expected Behavior**:
The `Spacingd` transformation should correctly handle spacing adjustments without causing any unexpected artifacts.

Thanks in advanced!

**Comment by KumoLiu** [comment]
Hi @emi-dm, I ran the notebook you shared, and found the problem is from the `Orientationd`.
After your `Orientation`, the affine will be wrong. 
![Screen Shot 2024-02-19 at 13 50 59](https://github.com/Project-MONAI/MONAI/assets/55491388/d1418911-2f13-417c-b191-7573d289c96a)
In MONAI, we assume the affine matrix is `LPS` in `ITKReader` which should be your default reader.
https://github.com/Project-MONAI/MONAI/blob/ff430286c37e78d7592372a5a97377f0cbb0219c/monai/data/image_reader.py#L176

Hope it helps, thanks.
https://github.com/Project-MONAI/MONAI/blob/ff430286c37e78d7592372a5a97377f0cbb0219c/monai/data/image_reader.py#L282

**Reply by emi-dm** [reply]
Hi @KumoLiu, can you modify my notebook and show me how i can solve it? Thanks in advanced!

**Reply by KumoLiu** [reply]
I saw you modified the notebook and removed the `Orientationd`, is there still an issue exists?

**Reply by emi-dm** [reply]
Yes, I would like to use Spacingd and Orientationd transformations to standardize my images. These have a base resolution of 512x512 px, I would like to normalize all volumes to the same spacing and finally have a common resolution (e.g. 256x256 px). 

I restored the original version of the notebook.

Thanks in advanced!!

**Reply by KumoLiu** [reply]
Hi @emi-dm, I tried with larger pixdim, and then the kernel will not crash. So the problem should be the limited RAM resource.
`monai.transforms.Spacingd(keys=["image"], pixdim=[1.0, 1.0, 1.])`

Thanks.

**Reply by emi-dm** [reply]
Thank you very much, it is true that by decreasing the spacing space I am generating more information, hence I run out of RAM. 

Thanks for all the information and for giving me a hand!

**Reply by emi-dm** [reply]
@KumoLiu , Do you know of any resources where I can learn more about this?
